### ISSUE TITLE
Unable to login to MP server after converting from 3.0.8 to 3.1.1

### ISSUE THREAD

**Issue by abortu** [DarkPacks/SevTech-Ages#3647]
## Bug Report

After upgrading the server to 3.1.1, I'm unable to login. No client or server crashes, just disconnects. I have the 3.0.8 and 3.1.1 (post conversion) world save files available on dropbox if they are needed. 

## Expected Behaviour

To be able to successfully login to the server. 

## Possible Solution

N/A

## Steps to Reproduce (for bugs)

1. Turn on server
2. Convert 3.0.8 world save to 3.1.1
3. Try to login

## Logs

* Client/Server Log:
https://gist.github.com/abortu/d30f78617e4df9975ba2e02654935998
* Crash Log:
N/A, no crashes occur. 

## Client Information

* Modpack Version: 3.1.1
* Java Version: 64-Bit Server VM, version 1.8.0_191, running on Linux:amd64:4.20.15-200.fc29.x86_64
* Launcher Used: MultiMC 0.6.4
* Memory Allocated: 8GB
* Server/LAN/Single Player:
MP server (local network)
* Optifine Installed: No
* Shaders Enabled: No

## World Information

* Modpack Version world created in: 3.0.8

* Additional Content Installed: N/A

## Server Information
* Java Version: OpenJDK 64-Bit Server VM, version 1.8.0_171, running on Linux:amd64:4.20.16-200.fc29.x86_64
* Operating System: Linux - Fedora 29
* Hoster/Hosting Solution: N/A
* Sponge (Non-Vanilla Forge) Server: N/A

**Comment by sk2048** [comment]
Hello, it appears your client log is incomplete so I'm not seeing if there's an error on the client side causing this - it only goes up to some point during init and does not include your connection attempts.

Your server log only flags that a time out is occurring during connection. Are you using the bundled startup script? If not you might consider adding the argument `-Dfml.readTimeout=180` to your environment. You can also try adding this to the client.

If this does not resolve your issue, you will probably find you get more and quicker support for this issue on the discord. https://discord.gg/darkosto

**Comment by abortu** [comment]
You need to view the raw version, gist truncates large files.

**Comment by abortu** [comment]
I'll try to add that option and report back.

**Comment by sk2048** [comment]
I tried that, besides it normally warns when it's truncating.

**Comment by abortu** [comment]
Weird, I just saw the raw version is messed up. I'll re-upload and send a link.

**Comment by abortu** [comment]
Re-uploaded client log link: https://gist.githubusercontent.com/abortu/fd04717afccdea04e3c5f03d7181c466/raw/df61fa3673436b239a8b8c91dc1496c9ad9288de/sevtech_311_client.log

**Comment by sk2048** [comment]
Yeah, your connection attempt is lasting exactly 30 seconds before being timed out and I'm not seeing anything else particularly worrying. I suspect the argument I mentioned above will solve this. If not, please do check out the discord, as I need sleep!

**Comment by abortu** [comment]
I added the argument to both client and server, it still fails. I'll hit up discord, thanks for the help!

New Logs:
https://gist.github.com/abortu/920c33249c2b770e3bcfc0903e559455

**Comment by abortu** [comment]
It appears this is an issue with the server running on OpenJDK 1.8 under Linux. I'm still testing and will submit my findings once I'm done. Switching to Oracle's java has let me login. I tried this locally and and am now going to test this on my server to see if the results are the same.

**Comment by abortu** [comment]
I'm able to login now. Here are the steps I tried and the results.

Removed Docker from the equation, ran server as usual (ServerStart.sh)

OpenJDK 1.8_201: Same behavior, unable to connect, no errors. Just looks like I connected and disconnected normally.

Oracle 1.8_201: 
  - Running on the already converted save (done by OpenJDK): logged in, but lost all of my stage information.
  - Re ran the conversion process using a fresh copy of my 3.0.8 save. Connects and staging appears to be correct now. No manually giving stages needed.

Re-ran using docker (with Oracle 1.8)
  - Ran as intended, connects, staging is fine. 

Thanks for all the help on here and discord! I'll close this issue, just putting the info out there in case someone comes across a similar situation.
